src/types/BulkUpload.ts

```diff
@@ -0,0 +1,42 @@
+import { ajv } from '../ajv';
+import type { JSONSchemaType } from 'ajv';
+
+export enum BulkUploadStatus {
+  PENDING = 'pending',
+  IN_PROGRESS = 'in_progress',
+  COMPLETED = 'completed',
+  FAILED = 'failed',
+  CANCELED = 'canceled',
+}
+
+export interface BulkUpload {
+  readonly id: number;
+  fileName: string
+  sourceUrl: string;
+  destinationUrl: string;
```

Comment:
`destinationUrl` is generic, and the protocol of the URL could be part of the url content (e.g. `S3://` or `SFTP://` or `HTTP://`).  If we want to force S3, though, then maybe we shouldn't be storing a URL at all but instead should be storing the relevant S3 pieces directly (key? bucket? region?).

I suppose if we were baking granular S3 components into the schema we'd want:

1. DestinationBucket
2. DestinationKey
3. region too?  Maybe?

Reply:
Can you say more about what you intend `destinationUrl` to be used for? I don't remember any conversations we might have had about it, and I'm not able to infer it from context. My best guess is that this is will be the stable URL of the uploaded file after it's processed - is that your intention? Why is that useful to record here?

May I suggest dropping it from this PR, and adding it later if & when we're ready to populate it?

Reply:
Yeah I'll drop it from this PR (though to the second question of "why here" -- this is the only persistent record of a bulk upload, so I think here is the best -- but we can cross that bridge when it's implemented, maybe there will be a better choice at that point!)